src/irr/asset/CCPUMeshPacker.h

```diff
@@ -15,14 +16,15 @@ template <typename MDIStructType = DrawElementsIndirectCommand_t>
 class CCPUMeshPacker final : public IMeshPacker<ICPUMeshBuffer, MDIStructType>
 {
 public:
-	CCPUMeshPacker(const SVertexInputParams& preDefinedLayout, const AllocationParams& allocParams, uint16_t maxIndexCountPerMDIData = std::numeric_limits<uint16_t>::max())
-		:IMeshPacker<ICPUMeshBuffer, MDIStructType>(preDefinedLayout, allocParams, maxIndexCountPerMDIData) 
+	CCPUMeshPacker(const SVertexInputParams& preDefinedLayout, const AllocationParams& allocParams, uint16_t minTriangleCountPerMDIData = 256u, uint16_t maxTriangleCountPerMDIData = 1024u)
+		:IMeshPacker<ICPUMeshBuffer, MDIStructType>(preDefinedLayout, allocParams, minTriangleCountPerMDIData, maxTriangleCountPerMDIData)
 	{
 		m_outMDIData = core::make_smart_refctd_ptr<ICPUBuffer>(allocParams.MDIDataBuffSupportedSizeInBytes);
 		m_outIdxBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(allocParams.indexBuffSupportedSizeInBytes);
 	}
 
-	virtual PackedMeshBufferData alloc(const ICPUMeshBuffer const* meshBuffer) override;
+	template <typename Iterator>
+	ReservedAllocationMeshBuffers alloc(const Iterator begin, const Iterator end);
```

Comment:
move this to parent, and have it call a virtual `alloc_do_xxxxx` for every divergent (different for CPU or GPU mesh) thing you need to do

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/496#discussion_r443521117